### PR TITLE
Add OpenAI Plugin Example

### PR DESCRIPTION
**Pull request by vulus98**
- [ ] Address the latest comments on the README

### CODE REVIEW

File: examples/openai/README.md
Comment (davidoort) [inline examples/openai/README.md:4]: please rebase on master and keep this as npm run start

**Review by davidoort — CHANGES_REQUESTED** [review]
Please rebase on master and make sure lib files are gitignored

File: plugins/openai/README.md
Comment (davidoort) [inline plugins/openai/README.md:27]: This is clearly WIP, needs to be completed before merging at least with a couple of examples

**Comment by vulus98** [comment]
Filled in the readme both for plugin and for examples

File: examples/openai/.gitignore
Comment (davidoort) [inline examples/openai/.gitignore:2]: is this necessary? I think they should probably be included in the top level of the repo (if not already in main branch feel free to add it there)

File: examples/openai/README.md
Comment (davidoort) [inline examples/openai/README.md:18]: for conciseness we can even skip this as the plugin will fetch it from this env variable automatically. So we can just put openAI()

File: examples/openai/README.md
Comment (davidoort) [inline examples/openai/README.md:298]: in order to prevent this from getting buried, I'd consider putting the usage examples in a separate doc that gets linked in the examples/openai/README.md file

This will also be useful if we move to a structure where we have examples for all plugins in one package

File: plugins/openai/README.md
Comment (davidoort) [inline plugins/openai/README.md:104]: this is good, I'd keep the list of examples even shorter or people will not even see this. Pick max 2 examples. The genkit docs link should be https://firebase.google.com/docs/genkit/get-started

Also, make sure to directly link to the README file of the examples for openai specifically

Also, skip the import statements in these examples for brevity

**Review by davidoort — CHANGES_REQUESTED** [review]
Great work on the docs!

I added some small comments here and there, and don't forget to address the old ones 🙏

Did not dig too deep into the individual examples because it doesn't need to be perfect for now

File: examples/openai/README.md
Comment (vulus98) [reply examples/openai/README.md]: I'll check, but yes,it should work automatically

File: examples/openai/README.md
Comment (vulus98) [reply examples/openai/README.md]: I couldn't find the markdown support for it, only some external tools that allow that. If I got you right, you want the same content, but would like to separate the usage examples in the new markdown file which gets imported in our README.md?

Comment (vulus98) [reply examples/openai/README.md]: Or to omit the examples in the main README and just link to other .md file which will contain them?

File: .gitignore
Comment (davidoort) [inline .gitignore:6]: not needed I think. node_modules/ I think includes all instances of node_modules in the repo, but please let me know if I'm wrong

File: examples/README.md
Comment (davidoort) [inline examples/README.md:26]: should be lowercase

File: examples/src/index.ts
Comment (davidoort) [inline examples/src/index.ts:15]: we should import all plugins here so that the DevUI shows all of them

File: plugins/openai/src/gpt.ts
Comment (davidoort) [inline plugins/openai/src/gpt.ts:67]: let's follow existing conventions (camelCase)

File: plugins/openai/src/gpt.ts
Comment (davidoort) [inline plugins/openai/src/gpt.ts:167]: ideally all this plugin work would be in a separate PR, but I can let it pass for now

**Review by davidoort — CHANGES_REQUESTED** [review]
small things

File: examples/openai/README.md
Comment (davidoort) [reply examples/openai/README.md]: Why not: [Usage examples OpenAI Plugin](docs/usage_openai.md)?
I'd be in favor of the latter. so we can link to docs/usage_openai.md and other ones. lmk if this makes sense

**Comment by vulus98** [comment]
Check currently fail as it checks if the genkit version is 11, but we updated it to 12 in the code. I'll leave it to you to change this test and be aware of this change before merging

**Review by davidoort — APPROVED** [review]

**Pull request merged**
